src/prefect/engine/result_handlers/pandas_result_handler.py

```diff
@@ -0,0 +1,148 @@
+import pandas as pd
+import os
+import typing
+
+from prefect.engine.result_handlers.result_handler import ResultHandler
+
+
+class PandasResultHandler(ResultHandler):
+    """
+    Hook for storing and retrieving task results in Pandas DataFrames.
+    Task results are written/read via the standard Pandas
+    ``to_[FILETYPE]``/``read_[FILETYPE]`` methods, and additionally the filename
+    can be fully specified at task instantiation.
+
+    Args:
+        - path (str): Filepath to be read from or written to, including file name and extension.
+        - file_type (str): The type of file to write to, e.g. "csv" or "parquet". Must match the name
+            of the appropriate ``to_[FILETYPE]``/``read_[FILETYPE]`` method.
+        - read_kwargs (dict, optional): If present, passed as **kwargs to the ``read_[FILETYPE]`` method.
+        - write_kwargs (dict, optional): If present, passed as **kwargs to the ``to_[FILETYPE]`` method.
+
+    .. note::
+        Because the filepath is fully specified, when using this handler in a ``map``
+        can lead to the same file being read to or written from by every iteration of
+        the map. To deal with this case this handler allows for Python's ``str.format``
+        to be used with the names of arguments to the task instance. For instance, if
+        your task has an argument named ``sample_name`` that you intend to map over,
```

Comment:
This is a great idea! We _might_ consider expanding this as a general rule to also allow for formatting from keys in `prefect.context`